expression/builtin_arithmetic.go

```diff
@@ -453,6 +455,56 @@ func (s *builtinArithmeticMultiplyIntSig) evalInt(row []types.Datum) (val int64,
 	return result, false, nil
 }
 
+type arithmeticIntDivideFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *arithmeticIntDivideFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpInt, tpDecimal, tpDecimal)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	setFlenDecimal4Int(bf.tp, args[0].GetType(), args[1].GetType())
```

Comment:
use `setFlenDecimal4Int` to set Flen maybe is not correct ? you can test whether the `Flen` and `Decimal` is settled properly by start your mysql client with `--column-type-info` options and check the `Length` and 
`Decimal` value.

```sql
drop table if exists t;
create table t(a bigint, b double);
insert into t select 12, 2.009;
```

for now, TiDB returns: 
```sql
TiDB > select a div b from t;
Field   1:  `a div b`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       LONGLONG
Collation:  binary (63)
Length:     0
Max_length: 1
Decimals:   31
Flags:      BINARY NUM


+---------+
| a div b |
+---------+
|       5 |
+---------+
1 row in set (0.01 sec)
```

and MySQL returns:
```sql
MySQL > select a div b from t;
Field   1:  `a div b`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       LONGLONG
Collation:  binary (63)
Length:     20
Max_length: 1
Decimals:   0
Flags:      BINARY NUM


+---------+
| a div b |
+---------+
|       5 |
+---------+
1 row in set (0.00 sec)
```